mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -1598,6 +1604,27 @@ private ScheduleRequest createSchedulingRequest(
             }
         }
 
+        /**
+         * Merges attributes assignment between job and artifact definitions. It does it by first fetching
+         * the associated JobArtifact tags using the artifact ID from the job, and then merging them with the assignment
+         * attributes from the job definition itself. The keys from the job definition take precedence over the
+         * keys from the artifact's tags.
+         *
+         * @param artifactID the artifact used by the job whose attributes are to be fetched
+         * @return A merged map of assignment attributes. The precedence of keys follows: job definition > artifact's tags.
+         */
+        private Map<String, String> mergeJobDefAndArtifactAssigmentAttributes(ArtifactID artifactID) {
```

Comment:
should this happen and persist at the job cluster level so the user can get the result of the final attributes?

Reply:
It could be done but I don't know how worth it is given that the final attributes are compute at job submission time anyway since some attributes can still be provided at that time.

Reply:
I feel like we want to display some result back to users so it's visible to them what is actually being used. Is there any info can be used to show such info on UI?